File: IAIP.Testx/UnitTestTests.cs
using Xunit;

namespace IAIP.Testx
{
    public class UnitTestTests
    {
        [Fact]
        private void PassingTest() => Assert.Equal(4, Add(2, 2));

        //[Fact]
        //private void FailingTest() => Assert.Equal(5, Add(2, 2));

        [Theory]
        [InlineData(1)]
        [InlineData(2)]
        [InlineData(3)]
        [InlineData(4)]
        private void TestOddEven(int value)
        {
            if (value % 2 == 0)
            {
                Assert.False(IsOdd(value));
                Assert.True(IsEven(value));
            }
            else
            {
                Assert.True(IsOdd(value));
                Assert.False(IsEven(value));
            }
        }

        private int Add(int x, int y) => x + y;

        private bool IsOdd(int x) => x % 2 == 1;

        private bool IsEven(int x) => x % 2 == 0;
    }
}

File: IAIP.Test/UnitTestTests.vb
Imports Xunit

Public Class UnitTestTests

    <Fact>
    Private Sub PassingTest()
        Assert.Equal(4, Add(2, 2))
    End Sub

    '<Fact>
    'Private Sub FailingTest()
    '    Assert.Equal(5, Add(2, 2))
    'End Sub

    <Theory>
    <InlineData(1)>
    <InlineData(2)>
    <InlineData(3)>
    <InlineData(4)>
    Private Sub TestOddEven(value As Integer)
        If value Mod 2 = 0 Then
            Assert.False(IsOdd(value))
            Assert.True(IsEven(value))
        Else
            Assert.True(IsOdd(value))
            Assert.False(IsEven(value))
        End If

    End Sub

    Private Shared Function Add(x As Integer, y As Integer) As Integer
        Return x + y
    End Function

    Private Shared Function IsOdd(x As Integer) As Boolean
        Return x Mod 2 = 1
    End Function

    Private Shared Function IsEven(x As Integer) As Boolean
        Return x Mod 2 = 0
    End Function

End Class
